docs/BuildFromSource.md

```diff
@@ -1,151 +1,185 @@
+Building the repo requires version 10.14.2 or newer of Node. You can find installation executables for Node at https://nodejs.org.
 
-Opening solution filters and building may produce an error code CS0006 with a message such
+**[Yarn](https://yarnpkg.com/)**
 
-> Error CS0006 Metadata file 'C:\src\aspnet\AspNetCore\artifacts\bin\Microsoft.AspNetCore.Metadata\Debug\netstandard2.0\Microsoft.AspNetCore.Metadata.dll' could not be found
+NodeJS installes the Node package manager (npm) by default. This repo depends on yarn, an alternate package manager for the Node ecosystem. You can install Yarn from the command line using the following command.
 
-The cause of this problem is that the solution filter you are using does not include the project that produces this .dll. This most often occurs after we have added new projects to the repo, but failed to update our .sln/slnf files to include the new project. In some cases, it is sometimes the intended behavior of the .slnf which has been crafted to only include a subset of projects.
+```bash
+$ npm install -g yarn
+```
 
-#### You can fix this in one of three ways
+**Java Development Kit**
 
-1. Build the project on command line. In most cases, running `build.cmd` on command line solves this problem.
-2. If the project is missing from the .sln file entirely, you can use `dotnet sln add` to add it, or else right click on the solution/folder in Visual Studio and choose Add->Existing Project, and adding it.
-3. If it is present in the .sln, but not the .slnf, you can update the solution filter to include the missing project. You can either do this one by right-clicking on project in Visual Studio and choosing to load it's direct dependencies, and then saving.  Alternatively, you can hand edit the .slnf file - it's a fairly simple json format.
+This repo contains some Java source code that depends on an install of the JDK v11 or newer. The JDK can be installed from either:
+  * OpenJDK <https://jdk.java.net/>
+  * Oracle's JDK <https://www.oracle.com/technetwork/java/javase/downloads/index.html>
 
-### Common error: Unable to locate the .NET Core SDK
+Similar to the instructions above for Windows, be sure that the the `JAVA_HOME` environment variable is set to the location of your Java installation.
 
-Executing `.\restore.cmd` or `.\build.cmd` may produce these errors:
+## Step 3: Build the repo
 
-> error : Unable to locate the .NET Core SDK. Check that it is installed and that the version specified in global.json (if any) matches the installed version.
-> error MSB4236: The SDK 'Microsoft.NET.Sdk' specified could not be found.
+Before opening our .sln/.slnf files in Visual Studio or VS Code, you will need to build the repo locally.
 
-In most cases, this is because the option _Use previews of the .NET Core SDK_ in VS2019 is not checked. Start Visual Studio, go to _Tools > Options_ and check _Use previews of the .NET Core SDK_ under _Environment > Preview Features_.
+### On Visual Studio
```

Comment:
I don't know why but I usually hear "from the command line" or "on the command line" but always "in Visual Studio".
```suggestion
### In Visual Studio
```